mobile/src/components/course/course.tsx

```diff
@@ -52,13 +68,30 @@ const Course: FC = () => {
     userPermissions: user?.permissions ?? [],
   });
 
+  const toggleSeeMore = (): void => {
+    setIsSeeMoreShown(!isSeeMoreShown);
+  };
+
+  useEffect(() => {
+    if (course && course.description) {
+      const descriptionText = getTextWithoutHTMLTags(course.description);
+
+      if (
+        descriptionText.split(' ').length >
+        CourseDescriptionFirstWordsCount.DEFAULT_COUNT
+      ) {
+        setIsSeeMoreShown(true);
+      }
+    }
+  }, [course]);
```

Comment:
I think we don't need this, if you want to reset it use useFocusEffect and pass setIsDescriptionExpanded(false) in return. Also check about is the length enough to show toggler should be inside CourseDescription component